src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -2013,30 +2036,40 @@ protected override void BeginProcessing()
                 }
             }
 
-            if (Wait.IsPresent)
+            if (Wait.IsPresent || _timeoutSpecified)
```

Comment:
If we'll use autogenerated property we can check Timeout != Threading.Timeout.Infinite

Reply:
Have changed as requested